modyn/models/articlenet/articlenet.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any
+
+import torch
+from torch import nn
+from transformers import DistilBertForSequenceClassification, DistilBertModel
+
+
+class ArticleNet:
+    # pylint: disable-next=unused-argument
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        self.model = ArticleNetwork(**model_configuration)
+        self.model.to(device)
+
+
+class DistilBertClassifier(DistilBertForSequenceClassification):
```

Comment:
what exactly does this do? I think without some documentation I am confused. You are overriding the main huggingface DistilBert classifier, not sure what the intention is here.

Reply:
I don't know exactly what happens here either. The code is taken from the benchmark library, it is not mine. See here https://github.com/huaxiuyao/Wild-Time/blob/main/wildtime/networks/article.py

Reply:
As mentioned in another comment, I think we should at least somewhat understand what is hapenning here and if that is truly doing what we want it to do. The problem with research code is that we build a chain of trust but nobody really understands what somebody else has written. It would be glad if we could figure this out since many thing can go wrong here with huggingface stuff.

Reply:
I have added some documentation. Shall we add unit tests? The problem is that it has to download the weights from huggingface

Reply:
For DistilBert, if we add tests, we should be able to somehow make the number of parameters small, like we did for DLRM (so it can be run on a CPU in testing) with random weights. If that is not possible, we should not add a test. For the other simpler models, we should add tests